azure-kusto-data/src/error.rs

```diff
@@ -37,22 +39,39 @@ pub enum Error {
     UnsupportedOperation(String),
 }
 
+/// Errors raised when an invalid argument or option is provided.
 #[derive(thiserror::Error, Debug, Clone, PartialEq, Eq)]
 pub enum InvalidArgumentError {
+    /// Error raised when a string denoting a duration is not valid.
     #[error("{0} is not a valid duration")]
     InvalidDuration(String),
+    /// Error raised when failing to convert a number to u32.
     #[error("{0} is too large to fit in a u32")]
     PayloadTooLarge(#[from] TryFromIntError),
 }
 
+/// Errors raised when parsing connection strings.
 #[derive(thiserror::Error, Debug, Clone, PartialEq, Eq)]
 pub enum ConnectionStringError {
+    /// Raised when a connection string is missing a required key.
     #[error("Missing value for key '{}'", key)]
-    MissingValue { key: String },
+    MissingValue {
+        /// The key that is missing.
+        key: String,
+    },
+    /// Raised when a connection string has an unexpected key.
     #[error("Unexpected key '{}'", key)]
-    UnexpectedKey { key: String },
+    UnexpectedKey {
+        /// The key that is unexpected.
+        key: String,
+    },
+    /// Raised when a connection string has an invalid value.
     #[error("Parsing error: {}", msg)]
-    ParsingError { msg: String },
+    ParsingError {
```

Comment:
should this just be `Parsing` to be consistent with other variant names?

Reply:
Good call!